astropy/utils/mixins.py

```diff
@@ -0,0 +1,71 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""A module containing miscellaneous mixin classes."""
+
+from typing import Mapping, Optional
+
+
+__all__ = ["DisplayFormatAttributeMixin"]
+
+
+class DisplayFormatAttributeMixin:
+    """Mixin class for storing format-specific representations of an object.
+
+    Attributes
+    ----------
+    _format : dict[str, str]
+        A mapping to format-specific representations of this object.
+        For example, for the ``Ohm`` unit in :mod:`astropy.units`, it might be
+        nice to have it displayed as ``\\Omega`` by the ``latex`` formatter.
+        In that case, `format` argument should be set to::
+
+            {'latex': r'\\Omega'}
+    name : str
+        The fallback value if a format-specific representation does not exist
+        in the `format` registry.
+
+    Methods
+    -------
+    get_format_name(format='latex')
+        Get a format-specific representation for this object.
+
+    Examples
+    --------
+    ``DisplayFormatAttributeMixin`` is a mixin class, so it must be subclassed.
+
+        >>> class Example(DisplayFormatAttributeMixin):
+        ...     def __init__(self, name, format=None):
+        ...         self.name = name
+        ...         self._format = format or {}  # always a dict
+
+        >>> ex = Example("example", format=dict(latex=r"ex$^{ample}$"))
+
+    Now we can get the format by name.
+
+        >>> ex.get_format_name("latex")
+        'ex$^{ample}$'
+
+    For non-defined formats, the default is the name of the instance.
+
+        >>> ex.get_format_name("not defined")
+        'example'
+    """
+
+    name: str
+    _format: Mapping[str, str]
+
+    def get_format_name(self, format: str) -> str:
+        """Get a format-specific representation for this object.
+
+        Uses the attributes ``_format`` and ``name``.
+
+        Parameters
+        ----------
+        format : str
+            The name of the format
+
+        Returns
+        -------
+        str
+            The representation of the object for the given format.
+        """
+        return self._format.get(format, self.name)
```

Comment:
Should this be `_name`?

Reply:
If we keep this method, I think it's better to stick to public API here. The Mixin expects a ``.name`` it's just an implementation detail that ``._name`` would also work in these cases.